source/Octo.Tests/Commands/CommandDesignFixture.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using FluentAssertions;
+using NUnit.Framework;
+using Octopus.Cli.Commands;
+using Octopus.Cli.Infrastructure;
+
+namespace Octo.Tests.Commands
+{
+    [TestFixture]
+    public class CommandDesignFixture
+    {
+        [Test]
+        public void ShouldAttachWithCorrectAttribute()
+        {
+            var commandClassNamesWithoutCorrectAttribute = Assembly.GetAssembly(typeof(ICommand)).GetTypes()
+                .Where(t => typeof(ICommand).IsAssignableFrom(t) &&
+                            t.GetCustomAttribute<CommandAttribute>() == null &&
+                            !t.IsAbstract &&
+                            !t.IsInterface)
+                .Select(t => t.Name)
+                .ToList()
+                .AsReadOnly();
+
+            commandClassNamesWithoutCorrectAttribute.Should().BeEmpty($"Each command which implements '{nameof(ICommand)}' interface must be attached with '{nameof(CommandAttribute)}'. " +
+                                                                      $"The following command classes: ({string.Join(", ", commandClassNamesWithoutCorrectAttribute)}) are not good enough for you?");
+        }
+
+        [Test]
+        public void SubClassesOfApiCommand_ShouldEitherOverrideExecuteMethodOrImplementCorrectInterface()
```

Comment:
What is the "CorrectInterface". This is a strange restriction and I can't figure out why by looking at this test.

Reply:
The `ApiCommand` will throw a run time exception if ... please refer to the code:
![image](https://user-images.githubusercontent.com/12688884/76173907-766aa500-61ef-11ea-9017-a686af4374fa.png)

Thought it will be nice to prevent it in the unit test rather than capturing this run-time error in test or even production environment?

Reply:
Yep. If this test fails on me though, can the test tell me more about how to fix it?

Reply:
yeah, it has the reason why it failed ->
`$"Each command which is a subclass of '{nameof(ApiCommand)}' class must only either override virtual '{methodName}' method Or implement {nameof(ISupportFormattedOutput)} interface. "`